include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -54,3 +54,67 @@ def VariableOp : MooreOp<"variable", [
     `` custom<ImplicitSSAName>($name) attr-dict `:` type($result)
   }];
 }
+
+def AlwaysCombOp : MooreOp<"alwayscomb", [
+  SingleBlock,
+  NoTerminator,
+  NoRegionArguments,
+  RecursiveMemoryEffects,
+  RecursivelySpeculatable
+]> {
+  let summary = "'alwayscomb block";
+  let description = "See SV Spec 9.2, and 9.2.2.2.";
```

Comment:
Cool! If I read "9.2 Structured procedures" in the spec correctly, the procedures `initial`, `always`, `always_comb`, `always_latch`, and `always_ff` all have identical structure: keyword and the statement to execute. It might make sense to model this as one `moore.procedure` op that has a `kind` attribute to select which one it is. Similar to https://github.com/llvm/circt/blob/326e507b7390a16b1b05f470fd24105f4ccb0b9c/include/circt/Dialect/Seq/SeqAttributes.td#L18-L23.

That would allow us to treat the procedures identically where possible, and only distinguish between the different flavors when we do actual lowering.

Reply:
Yeah, I notice that. Hence, I put them together, which can be reviewed in the Structure.cpp at lines 139 to 168. And I guess I will spend much time defining them in the MooreOps.td file. Because I am not good at writing td.

Reply:
I try to model this as one moore.procedure op, but I failed. I will keep up the good work.